File: mvvmsmart/src/main/java/com/wzq/mvvmsmart/base/IBaseViewModelMVVM.java
package com.wzq.mvvmsmart.base;


import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.OnLifecycleEvent;

/**
 * 这是一个接口，用于将一个类标记为LifecycleObserver
 * 接口里面没有任何方法，而是依赖于 {@lOnLifecycleEvent}注释方法
 * BaseViewModelMVVM ,实现了这个接口,所有具有生命周期感知
 */
public interface IBaseViewModelMVVM extends LifecycleObserver {

    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
    void onAny(LifecycleOwner owner, Lifecycle.Event event);

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    void onCreate();

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    void onDestroy();

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    void onStart();

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    void onStop();

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    void onResume();

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    void onPause();
}

File: mvvmsmart/src/main/java/com/wzq/mvvmsmart/base/IBaseViewModelMVVM.kt
package com.wzq.mvvmsmart.base

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.OnLifecycleEvent

/**
 * 这是一个接口，用于将一个类标记为LifecycleObserver
 * 接口里面没有任何方法，而是依赖于 {@lOnLifecycleEvent}注释方法
 * BaseViewModelMVVM ,实现了这个接口,所有具有生命周期感知
 */
interface IBaseViewModelMVVM : LifecycleObserver {
    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
    fun onAny(owner: LifecycleOwner?, event: Lifecycle.Event?)

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun onCreate()

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    fun onDestroy()

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun onStart()

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun onStop()

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    fun onResume()

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun onPause()
}